File: src-tauri/src/commands/twitch/eventsub.rs
use crate::services;
use crate::services::twitch::eventsub::EventKind;
use serde::Deserialize;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SubscribeParams {
    pub broadcaster_id: String,
    pub kind: EventKind,
}

#[tauri::command]
pub async fn subscribe(app: tauri::AppHandle, params: SubscribeParams) -> Result<(), String> {
    services::twitch::eventsub::subscribe(&app, params.broadcaster_id, params.kind).await
}

#[tauri::command]
pub async fn unsubscribe(app: tauri::AppHandle, params: SubscribeParams) -> Result<(), String> {
    services::twitch::eventsub::unsubscribe(&app, params.broadcaster_id, params.kind).await
}

File: src/lib/api/twitch/eventsub.ts
import type { EventKind } from "../../types/twitch/eventsub.ts";
import { invokeCommand, type InvokeOptions } from "../utils.ts";

export type SubscribeParams = { broadcasterId: string; kind: EventKind };

export async function subscribe(
  params: SubscribeParams,
  options?: InvokeOptions,
): Promise<void> {
  await invokeCommand("subscribe", params, options);
}

export async function unsubscribe(
  params: SubscribeParams,
  options?: InvokeOptions,
): Promise<void> {
  await invokeCommand("unsubscribe", params, options);
}
